### ISSUE TITLE
the library is not safe for parallelism

### ISSUE THREAD

**Issue by turtleDev** [sajari/docconv#141]
hey folks.

docconv is a great library. It solved a major problem for me, yet it has one glaring draw back: parallelism. 

the library can get deadlocked if there's an error (https://github.com/sajari/docconv/issues/112), or you can run into
a concurrent map write problem if you're using the library as a part of a worker pool.

I'm using this library in a small tool that uses multiple goroutines to parse and process files in a fan-out pattern. Whenever I use N = cores available, I run into one or the other panic due to dependencies themselves not being reentrant. (msoleps.Reader() is not reentrant)

```
fatal error: concurrent map read and map write

goroutine 90426 [running]:
github.com/richardlehane/msoleps.(*Reader).start(0xc0017a9f78, {0x939920?, 0xc010c82600?})
	/home/turtledev/go/pkg/mod/github.com/richardlehane/msoleps@v1.0.3/msoleps.go:128 +0x51a
github.com/richardlehane/msoleps.(*Reader).Reset(0x939960?, {0x939920?, 0xc010c82600?})
	/home/turtledev/go/pkg/mod/github.com/richardlehane/msoleps@v1.0.3/msoleps.go:76 +0x3b
github.com/turtledev/docconv.ConvertDoc.func1()
	/home/turtledev/go/pkg/mod/github.com/turtledev/docconv@v0.0.2-0.20230809133855-88aa796ecd5e/doc.go:46 +0x18c
created by github.com/turtledev/docconv.ConvertDoc
	/home/turtledev/go/pkg/mod/github.com/turtledev/docconv@v0.0.2-0.20230809133855-88aa796ecd5e/doc.go:27 +0x198
```

Maybe we should add a disclaimer somewhere warning users that this library is not safe for use in concurrent applications?

**Comment by jonathaningram** [comment]
@turtleDev hey, we just fixed #112. You can try the latest version and let us know if you find any other concurrent issues. We'll try our best to get any fixes in.

**Comment by turtleDev** [comment]
Hey thanks! I took it for a spin and it works.
